lib/controller.dart

```diff
@@ -38,19 +59,61 @@ void init() async {
 }
 
 void onLoginCompleted() async {
+  view.showLoading();
+
   await loadDataFromFirebase();
+  _filteredInteractions = _allInteractions;
+  _filteredComparisonInteractions = _allInteractions;
+
+  _uniqueFieldValues =
+      computeUniqFieldValues(_config.filters, _allInteractions);
+  _selectedTab = _config.tabs.first.id;
+
   view.setNavlinkSelected(_currentNavLink);
   _navLinks['analyse'].render();
+
+  view.hideLoading();
 }
 
 void onLogoutCompleted() async {
   logger.debug('Delete all local data');
 }
 
 void loadDataFromFirebase() async {
-  view.showLoading();
-  logger.debug('Read all required data');
-  view.hideLoading();
+  try {
+    _config = await fb.fetchConfig();
+  } catch (e) {
+    view.showAlert(_errorMessages['parse-config']);
+    logger.error(e);
+    rethrow;
+  }
+
+  try {
+    _allInteractions =
+        await fb.fetchInteractions(_config.data_paths['interactions']);
+  } catch (e) {
+    view.showAlert(_errorMessages['fetch-interactions']);
+    logger.error(e);
+    rethrow;
+  }
+}
+
+Map<String, Set> computeUniqFieldValues(List<model.Filter> filterOptions,
+    Map<String, Map<String, dynamic>> interactions) {
+  var uniqueFieldValues;
```

Comment:
I'm not sure how this works without throwing a null-related error in the lines below. I think it needs to be instantiated first?
```suggestion
  Map<String, Set> uniqueFieldValues = {};
```

Reply:
Fixed from the PR #7